File: metodos/Gaus Simple.py
import numpy as np

def GausSimple(A, b):
    # Inicialización
    n = A.shape[0]
    M = np.concatenate((A, b.reshape(n, 1)), axis=1)
    
    # Reducción del sistema
    for i in range(n-1):
        for j in range(i+1, n):
            if M[j, i] != 0:
                M[j, i:n+1] = M[j, i:n+1] - (M[j, i]/M[i, i]) * M[i, i:n+1]
    
    # Sustitución regresiva
    x = SustitucionRegresiva(M)
    
    # Entrega de resultados
    return x

def SustitucionRegresiva(M):
    n = M.shape[0]
    x = np.zeros(n)
    for i in range(n-1, -1, -1):
        x[i] = (M[i, n] - np.dot(M[i, i+1:n], x[i+1:n])) / M[i, i]
    return x

File: metodos/GausSimple.java
import java.util.Arrays;

public class GaussianEliminationSimple {
    public static double[] gaussianEliminationSimple(double[][] A, double[] b) {
        int n = A.length;
        double[][] M = new double[n][n + 1];
        double[] x = new double[n];

        // Inicialización de la matriz extendida M
        for (int i = 0; i < n; i++) {
            System.arraycopy(A[i], 0, M[i], 0, n);
            M[i][n] = b[i];
        }

        // Reducción del sistema
        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                if (M[j][i] != 0) {
                    double factor = M[j][i] / M[i][i];
                    for (int k = i; k <= n; k++) {
                        M[j][k] -= factor * M[i][k];
                    }
                }
            }
        }

        // Sustitución regresiva
        for (int i = n - 1; i >= 0; i--) {
            double dotProduct = 0.0;
            for (int j = i + 1; j < n; j++) {
                dotProduct += M[i][j] * x[j];
            }
            x[i] = (M[i][n] - dotProduct) / M[i][i];
        }

        return x;
    }
}
